### PR TITLE
Fix master branch cargo test fail

### LINKED ISSUE #4253: Cargo test fail
**Describe the bug**
run cargo test
```
error[E0609]: no field `alias` on type `&datafusion_expr::Union`
   --> datafusion/optimizer/src/propagate_empty_relation.rs:123:35
    |
123 | ...                   union.alias.clone(),
    |                             ^^^^^ unknown field
    |
    = note: available fields are: `inputs`, `schema`

```
Seems due to two recent two commits
https://github.com/apache/arrow-datafusion/commit/5de9709fadf1344baca1ed3e1f9a44e06dfa9a63
and 
https://github.com/apache/arrow-datafusion/commit/4dcf9857c0aad3d11050037bd46889a54f83a394

@jackwener  could mind take a look ?

**To Reproduce**

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by Ted-Jiang**
Signed-off-by: yangjiang <yangjiang@ebay.com>

# Which issue does this PR close?

Closes #4253 .

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/optimizer/src/propagate_empty_relation.rs [hunk 1]
```diff
@@ -120,14 +120,13 @@ impl OptimizerRule for PropagateEmptyRelation {
                         Ok(LogicalPlan::Projection(Projection::new_from_schema(
                             Arc::new(child),
                             optimized_children_plan.schema().clone(),
-                            union.alias.clone(),
+                            None
```
Comment (Ted-Jiang) [inline datafusion/optimizer/src/propagate_empty_relation.rs:123]: @jackwener  After  https://github.com/apache/arrow-datafusion/commit/4dcf9857c0aad3d11050037bd46889a54f83a394 
no field `alias` on `Union`, 

Is there ok pass `none` here?

**Comment by jackwener** [comment]
#4252 cc @Ted-Jiang.

File: datafusion/optimizer/src/propagate_empty_relation.rs (hunk 1 above)
Comment (jackwener) [reply datafusion/optimizer/src/propagate_empty_relation.rs]: It's due to `Bors` problem.😂

**Comment by Ted-Jiang** [comment]
dulicate https://github.com/apache/arrow-datafusion/pull/4252

**Pull request closed without merging**